### ISSUE TITLE
[TalkBack] Speak character deletion when "Keyboard echo" is turned off

### ISSUE THREAD

**Issue by GoogleCodeExporter** [clc/eyes-free#194]
```
What steps will reproduce the problem?
1. Open Settings->Accessibility->Talkback->Settings->Keyboard echo.
2. Set the value to "Never speaked typed keys".
3. Now open Messaging, create a new message and type some text into the 
message's text field.

What is the expected output? What do you see instead?
Expected: Deleting a character should put out what has been deleted via speech.
Result: Deletion of a character is silent.
What version of the product are you using? On what operating system?
Talkback 3.3.0_B3 and previous down to 3.2.1

Please provide any additional information below.
Happens with both German and English.

```

Original issue reported on code.google.com by `HGF...@gmail.com` on 28 Feb 2013 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
Sorry, I forgot to mention that this is with Android 4.2.2 on Nexus 4.
```

Original comment by `HGF...@gmail.com` on 28 Feb 2013 at 9:37

**Comment by GoogleCodeExporter** [comment]
Original comment by `alanv@google.com` on 6 Mar 2013 at 12:48
- Changed title: **[TalkBack] Speak character deletion when "Keyboard echo" is turned off**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
Original comment by `alanv@google.com` on 8 Mar 2013 at 7:25
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v3.3.1_b1.
```

Original comment by `alanv@google.com` on 3 Apr 2013 at 1:00
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks. However, it is only partially fixed.

Perform steps 1 to 3.
Type in a few words.
Delete character by character.
Result: When a space is deleted, TalkBack speaks the word before the just 
deleted space instead of "Space".
LG Nexus 4 with Android 4.2.2 and TalkBack 3.3.1_B2.
```

Original comment by `HGF...@gmail.com` on 8 Apr 2013 at 5:52
